modules/apps/commerce/commerce-discount-api/src/main/java/com/liferay/commerce/discount/application/strategy/CommerceDiscountApplicationStrategyRegistry.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.discount.application.strategy;
+
+/**
+ * @author Jiaxu Wei
+ */
+public interface CommerceDiscountApplicationStrategyRegistry {
+
+	public boolean containsKey(String commerceDiscountApplicationStrategyKey);
```

Comment:
@W2jx Please review the usages of `containsKey` method. If all the usages are followed by `get`, you can remove `containsKey`.

Just do

```
A a = registry.get(key);

if (a == null) {
    xxx;
}

// use a
```

Reply:
@W2jx By the way, even if there's usage of `containsKey` without following `get`, you should convert the usages of `containsKey` and `get` which are used together to the form above.

Reply:
@W2jx And I think all the `containsKey` usages can actually replaced with `get` + null check.